packages/server/src/fhir/repo.ts

```diff
@@ -1688,41 +1697,52 @@ export class Repository extends FhirRepository<PoolClient> implements Disposable
    * @param create - Flag for when "creating" vs "updating".
    * @returns The account value.
    */
-  private async getAccount(
+  private async getAccounts(
     existing: Resource | undefined,
     updated: Resource,
     create: boolean
-  ): Promise<Reference | undefined> {
-    const account = updated.meta?.account;
-    if (account && this.canWriteAccount()) {
+  ): Promise<Reference[] | undefined> {
+    const updatedAccounts = this.extractAccountReferences(updated.meta);
+    if (updatedAccounts && this.canWriteAccount()) {
       // If the user specifies an account, allow it if they have permission.
-      return account;
+      return updatedAccounts;
     }
 
     if (create && this.context.accessPolicy?.compartment) {
       // If the user access policy specifies a compartment, then use it as the account.
-      return this.context.accessPolicy.compartment;
+      return [this.context.accessPolicy.compartment];
     }
 
     if (updated.resourceType !== 'Patient') {
       for (const patientRef of getPatients(updated)) {
-        // If the resource is in a patient compartment, then lookup the patient.
         try {
-          const systemRepo = getSystemRepo();
-          const patient = await systemRepo.readReference(patientRef);
-          if (patient.meta?.account) {
-            // If the patient has an account, then use it as the resource account.
-            return patient.meta.account;
-          }
+          // If the resource is in a patient compartment, then lookup the patient.
+          const patient = await getSystemRepo().readReference(patientRef);
+          // If the patient has an account, then use it as the resource account.
+          return this.extractAccountReferences(patient.meta);
         } catch (err: any) {
-          const ctx = getRequestContext();
-          ctx.logger.debug('Error setting patient compartment', err);
+          getRequestContext().logger.debug('Error setting patient compartment', err);
         }
       }
     }
 
     // Otherwise, default to the existing value.
-    return existing?.meta?.account ?? updated.meta?.account;
+    return this.extractAccountReferences(existing?.meta);
+  }
+
+  private extractAccountReferences(meta: Meta | undefined): Reference[] | undefined {
```

Comment:
Thoughts on using a `Set` here?

It seems like it might be possible for someone to include duplicates of the same value?